### ISSUE TITLE
Expander ignores repeat brackets, if DalSegnoAlCoda is involved.

### ISSUE THREAD

**Issue by seffka** [cuthbertLab/music21#557]
If a score includes both: repeats with brackets and Segno/AlCoda/DalSegnoAlCoda/Coda constructs:
![demo-1](https://user-images.githubusercontent.com/7113235/84305283-1a028300-ab5a-11ea-9227-0bd13c7f835f.png)

repeat brackets are ignored by Expander.

Desired situation: repeat brackets should work.

Suggested test case:

```
def testDalSegnoWithRepeats(self):
        from music21 import stream, note, bar

        p = stream.Part()
        m1 = stream.Measure()
        m1.append(note.Note('g4', type='whole'))
        m2 = stream.Measure()
        m2.append(Segno())
        m2.append(note.Note('a4', type='whole'))
        m2.append(Coda('to coda'))
        m3 = stream.Measure()
        m3.append(note.Note('b4', type='whole'))
        m4 = stream.Measure()
        m4.append(note.Note('d5', type='whole'))
        m4.append(DalSegnoAlCoda())
        m5 = stream.Measure()
        m5.append(Coda())
        m5.append(note.Note('c5', type='whole'))

        p.append([m1, m2, m3, m4, m5])
        m2.leftBarline = bar.Repeat(direction='start')
        rb1 = spanner.RepeatBracket(m3, number=1)
        m3.rightBarline = bar.Repeat()
        p.append(rb1)
        rb2 = spanner.RepeatBracket(m4, number=2)
        p.append(rb2)
        #p.show()

        ex = Expander(p)
        self.assertTrue(ex.isExpandable())
        post = ex.process()
        #FAIL!
        self.assertEqual(len(post), 7)
        self.assertEqual([n.name for n in post.flat.notes],
                         ['G', 'A', 'B', 'A', 'D','A', 'C'])
```

**Comment by mscuthbert** [comment]
thanks.  good test case.  hope i can get to it this summer -- a great test case usually makes the problem solvable in an hour or less.

it's esoteric in such a short piece, but totally traditional in longer songs, so should be fixed.

**Comment by seffka** [comment]
Thanks, Michael. This test case is quite clear. I want to post two more cases, which are rather ideas for future development.

**Comment by mscuthbert** [comment]
Yeah.  @seffka -- I'm realizing that there are three different Repeat marks relating to Codas ("to coda", "D.S. al coda", and the O sign itself).  And that music21 distinguishes them via their order rather than their content, which is something I'll need to fix.

Since I didn't write the repeat.Expander code, it'll take a little longer than usual, as I need to refactor and relearn as I go.

**Comment by seffka** [comment]
I see. Thank you!

**Comment by mscuthbert** [comment]
Still interested in this -- just haven't had the time.  But the really good test code makes me want to do it.

**Comment by seffka** [comment]
BTW, the last mentioned nuance of ToCoda and Repeats usage still can't be formally encoded in musicXML
 (https://github.com/w3c/musicxml/issues/322).
I.e., there could be specification: should repeats be effectively played on the way to coda or no.

I didn't re-approach this issue since then, the overall impression was that maybe some conceptual changes are required in the repeat module, so I haven't moved further than the test case.
